### PR TITLE
Fix the SmartRename button regressions

### PR DESCRIPTION
**Pull request by aoyangyou**
This PR fix the SmartRename button regressions: [https://devdiv.visualstudio.com/DevDiv/_workitems/edit/2112161/](url)
1. Suggestion panel behavior 
When `IsAutomaticSuggestionsEnabled` is ON
Button toggled => suggestion panel collapse
Button not toggled = > suggestion panel expand

2 Button UI when Hover / Focus / toggle / click 
Click => lighter border + purple background (indicate `IsAutomaticSuggestionsEnabled` is ON)
Focus => lighter border 
Toggle => lighter border + grey background
Hover => grey background
renamebuttoncolorRemoveBackground3

In blue and light theme
renamebuttoncolorRemoveBackgroundBlueLight

### CODE REVIEW

File: src/EditorFeatures/Core.Wpf/InlineRename/UI/SmartRename/SmartRenameUserInputComboBox.xaml [hunk 1]
```diff
@@ -87,7 +87,6 @@
                     </Trigger>
                     <Trigger Property="IsKeyboardFocused" Value="true">
                         <Setter Property="BorderBrush" Value="{DynamicResource {x:Static vsui:CommonControlsColors.ButtonBorderFocusedBrushKey}}"/>
-                        <Setter Property="Background" Value="{DynamicResource {x:Static vsui:CommonControlsColors.ButtonFocusedBrushKey}}"/>
```
Comment (Cosifne) [inline src/EditorFeatures/Core.Wpf/InlineRename/UI/SmartRename/SmartRenameUserInputComboBox.xaml:90]: Why this is not needed?

Comment (aoyangyou) [reply src/EditorFeatures/Core.Wpf/InlineRename/UI/SmartRename/SmartRenameUserInputComboBox.xaml:90]: We want to make it consistent with the panel expand case (Purple), and different with the panel collapse case. It decreases the confusion. 

After removing the focus background, the button will be purple when focused.  And the light border can indicate whether the button is focused.

Comment (Cosifne) [reply src/EditorFeatures/Core.Wpf/InlineRename/UI/SmartRename/SmartRenameUserInputComboBox.xaml:90]: Could you post a screenshot before & after?

Comment (Cosifne) [reply src/EditorFeatures/Core.Wpf/InlineRename/UI/SmartRename/SmartRenameUserInputComboBox.xaml:90]: Could you verify if the color still looks correct when in light/blue theme?

**Review by Cosifne — APPROVED** [review]

**Pull request merged**